### ISSUE TITLE
sbs.com.au ondemand has a new player and it detects ublock

### ISSUE THREAD

**Issue by KodeToad** [uBlockOrigin/uAssets#7836]
SBS Ondemand newly broken

### URL(s) where the issue occurs

`https://www.sbs.com.au/ondemand/`

### Describe the issue

Very recently (days), sbs has implemented a new player and now what worked perfectly before no longer works.
When trying to play any content containing ads, the player doesn't initiate and a message is displayed asking to disable ad-blocker.
If I disable ublock and re-enable it once the show starts, ads are displayed anyway

### Screenshot(s)

![screenshot-www sbs com au-2020 08 27-22_04_30](https://user-images.githubusercontent.com/3880336/91440234-a26b0700-e8b1-11ea-8262-b72589d6f8d2.png)

### Versions

- Browser/version: Chrome Version 85.0.4183.83 (Official Build) (64-bit)
- uBlock Origin version: 1.29.2

### Settings

N/A

### Notes

It was working a few days ago until sbs ondemand announced a new look player on their main page.
I tried disabling ublock and starting the player, but that didn't work. The ad played anyway.
It needs a free account to watch (any email is good for signup), and may need a proxy or vpn to connect to the australian site.
I'm not sure if this issue affects any other sbs site

**Comment by okiehsch** [comment]
````
@@||imasdk.googleapis.com/js/sdkloader/*$script,domain=sbs.com.au
@@||pubads.g.doubleclick.net/ondemand/hls/content/*/streams$xhr,domain=sbs.com.au
````
fixes the Video Playback Error on my end.
Does the video play fine for you if you add those filters?

**Comment by KodeToad** [comment]
That seems to fix the playback error, but the ads still play at the offset times, same as disabling and re-enabling ublock.
![screenshot-www sbs com au-2020 08 28-22_21_57](https://user-images.githubusercontent.com/3880336/91560368-16261600-e97d-11ea-955b-3520981da336.png)

**Comment by okiehsch** [comment]
Not sure if those can be entirely avoided without breaking the player.
Is it better if you also add
`||doubleclick.net/ondemand/hls/content/*/time-events.json$xhr,domain=sbs.com.au,important`

**Comment by KodeToad** [comment]
That doesn't seem to fix the problem ads.
Sad, I used to enjoy watching sbs.

**Comment by mapx-** [comment]
@KodeToad could you test

```
@@||sbs.com.au^$script,xhr,1p
sbs.com.au##+js(json-prune, ads)
```

**Comment by Poopooracoocoo** [comment]
It looks like the ads are inserted directly into the video when they're blocked!!! :(

**Comment by subby2006** [comment]
similar to #1905, sbs seems to be using dynamic ad insertation which is basically SSAI

**Comment by eplayerclick** [comment]
IF YOU  DO NOT WANT ADS STOP ENJOYING FREE CONTENT

**Comment by uBlock-user** [comment]
`no-xhr-if` scriptlet present in the dev build can address this issue.

**Comment by lain566** [comment]
> no-xhr-if scriptlet present in the dev build can address this issue.

The scriptlet does not work for me on `sbs.com.au`

<details><summary>Screenshot</summary>

![Captura de pantalla 2021-09-12 063948](https://user-images.githubusercontent.com/27441315/133132866-c2f3db1b-7069-4886-94b0-45239cff208a.png)
</details>

**Comment by lain566** [comment]
I just tested now

**Comment by lain566** [comment]
I tried adding `sbs.com.au##+js(no-xhr-if)` to see something logged in the console and it doesn't work either, but it works fine on `wetter.de`

**Comment by uBlock-user** [comment]
try with `sbs.com.au##+js(no-fetch-if)` ?

**Comment by eplayerclick** [comment]
stop sending me this rubbish

lun., 13 sept. 2021, 21:21 uBlock-user ***@***.***> a scris:

**Comment by lain566** [comment]
> try with sbs.com.au##+js(no-fetch-if) ?

they are not using `fetch`

**Comment by uBlock-user** [comment]
> https://github.com/orgs/uBlockOrigin/teams/ublock-filters-volunteers/discussions/234?from_comment=32#discussion-234-comment-32

There you said `no-xhr-if` worked fine, so you were testing with my scriptlet ?

**Comment by lain566** [comment]
Your scriptlet was working because the xhr request does not appear in the logger, however gorhill scriptlet does not work, the xhr request still appears in the logger

**Comment by uBlock-user** [comment]
@gorhill

**Comment by gorhill** [comment]
URL for me to investigate?

**Comment by uBlock-user** [comment]
`https://www.sbs.com.au/ondemand/watch/1936066627842`

**Comment by gorhill** [comment]
This requires AUS VPN and account, I will have to investigate later.

**Comment by gorhill** [comment]
What's the exact filter that was tried?

**Comment by lain566** [comment]
I have tried both
```
sbs.com.au##+js(no-xhr-if, pubads)
sbs.com.au##+js(no-xhr-if, doubleclick)
```
the request still appears in the logger, I am using Firefox, seems to work fine with Chrome.

**Comment by gorhill** [comment]
Maybe you do not have the scriptlet? Force an update of _"uBlock filters"_ to be sure.

**Comment by uBlock-user** [comment]
I reproduced the same issue yesterday too on Firefox, it's as if the site itself bypasses the scriptlet itself.

**Comment by uBlock-user** [comment]
I reproduced the same issue yesterday too on Firefox, it's as if the site itself bypasses the scriptlet.

**Comment by lain566** [comment]
> Maybe you do not have the scriptlet? Force an update of "uBlock filters" to be sure.

I had already done that several times.

**Comment by gorhill** [comment]
Does this work?

    sbs.com.au##+js(no-xhr-if, method:POST doubleclick)

**Comment by lain566** [comment]
It has worked only once, if I refresh the page it no longer works. This has happened to me before.

**Comment by krystian3w** [comment]
@[mapx-](https://github.com/mapx-) mapx- added the[ SSAI ](https://github.com/uBlockOrigin/uAssets/labels/SSAI)label [on 7 Apr 2018](https://github.com/uBlockOrigin/uAssets/issues/1905#event-1562292997)
@[jspenguin2017](https://github.com/jspenguin2017) [jspenguin2017](https://github.com/jspenguin2017) mentioned this issue on 9 May 2018
[Ads] cwtv.com NanoAdblocker/NanoFilters#25
Closed
@jspenguin2017 jspenguin2017 mentioned this issue on 3 Jun 2018
nbc.com #2409
Closed
@[okiehsch](https://github.com/okiehsch) okiehsch changed the title Cwtv video ads Websites using SSAI (server side ad injection) [on 3 Jun 2018](https://github.com/uBlockOrigin/uAssets/issues/1905#event-1659832690)
This was referenced on 3 Jun 2018
Viceland Video Ads #1541
Closed
uBO on Chrome not blocking fox.com #772
Closed
abc7.com #794
Closed
@[jspenguin2017](https://github.com/jspenguin2017) jspenguin2017 mentioned this issue on 22 Jul 2018
[Anti-adblock] investigationdiscovery.com #2966
Closed
@[SkeletalDemise](https://github.com/SkeletalDemise) SkeletalDemise mentioned this issue on 22 Jul 2018
anti Adblocker on discovery.com sciencechannel.com #2931
Closed
@[terrorist96](https://github.com/terrorist96) terrorist96 mentioned this issue on 2 Sep 2018
Ads Loading gorhill/uBO-Extra#76
Closed
@[jspenguin2017](https://github.com/jspenguin2017) jspenguin2017 mentioned this issue on 15 Sep 2018
Ads while watching shows on FOX #3486
Closed
@[uBlock-user](https://github.com/uBlock-user) uBlock-user mentioned this issue on 27 Oct 2018
Midroll Ads Aren't being blocked CityTV.com #3847
Closed
@[mapx-](https://github.com/mapx-) mapx- mentioned this issue on 17 Nov 2018
rediff.com | Tracking #4138
Closed
@[terrorist96](https://github.com/terrorist96) terrorist96 mentioned this issue on 23 Nov 2018
Fails to Block Ads on The CW - http://thecwtv.com #4179
Closed
@[stoically](https://github.com/stoically) stoically mentioned this issue on 20 Mar 2019
ALL twitch.tv issues #5184
Open
@[okiehsch](https://github.com/okiehsch) okiehsch mentioned this issue on 11 Jun 2019
https://www.brighteon.com/ #5803
Closed
@[llacb47](https://github.com/llacb47) llacb47 mentioned this issue on 17 Jun 2019
discovery.com videos not working easylist/easylist#3636
Closed
@https://github.com/uBlockOrigin uBlockOrigin deleted a comment from [prateekrastogi](https://github.com/prateekrastogi) [on 26 Jan 2020](https://github.com/uBlockOrigin/uAssets/issues/1905#event-2979959103)
@[uBlock-user](https://github.com/uBlock-user) uBlock-user added the[ can't fix ](https://github.com/uBlockOrigin/uAssets/labels/can%27t%20fix)label [on 17 Jun 2020](https://github.com/uBlockOrigin/uAssets/issues/1905#event-3453347081)
@[nebchan](https://github.com/nebchan) nebchan mentioned this issue on 29 Sep 2020
Twitch TV now injecting ads #7968
Closed
@[llacb47](https://github.com/llacb47) llacb47 mentioned this issue on 1 Nov 2020
Add queryprune filters to deal with SSAI #8131
Merged
@[uBlock-user](https://github.com/uBlock-user) uBlock-user mentioned this issue on 16 Aug 2021
pluto.tv: Ads not being blocked #9764
Closed
6 tasks
@[subby2006](https://github.com/subby2006) subby2006 mentioned this issue on 1 Sep 2021
sbs.com.au ondemand has a new player and it detects ublock #7836
Closed
[uBlock-user](https://github.com/uBlock-user) added a commit that referenced this issue on 20 Jan
@uBlock-user
[update](https://github.com/uBlockOrigin/uAssets/commit/eb9148f1e001b8862c044b8884a27c09a607fe1a) https://github.com/uBlockOrigin/uAssets/issues/1905#issuecomment-394129540
[eb9148f](https://github.com/uBlockOrigin/uAssets/commit/eb9148f1e001b8862c044b8884a27c09a607fe1a)
This was referenced on 21 Jan
go.tlc.com: ads #11419
Closed
go.discovery.com: ads #11420
Closed
@[uBlock-user](https://github.com/uBlock-user) uBlock-user closed this as [completed](https://github.com/uBlockOrigin/uAssets/issues?q=is%3Aissue+is%3Aclosed+archived%3Afalse+reason%3Acompleted) in https://github.com/uBlockOrigin/uAssets/commit/ac8860312f726292bd3e7c2f34a82f6d20614dbe [on 9 Feb](https://github.com/uBlockOrigin/uAssets/issues/1905#event-6035560189)
@[uBlock-user](https://github.com/uBlock-user) uBlock-user reopened this [on 9 Feb](https://github.com/uBlockOrigin/uAssets/issues/1905#event-6035564682)
[uBlock-user](https://github.com/uBlock-user) added a commit that referenced this issue on 9 Feb
@uBlock-user
[append](https://github.com/uBlockOrigin/uAssets/commit/fe89b1b850dc070f09914e36b5520e955a5ad3a6) https://github.com/uBlockOrigin/uAssets/issues/1905#issuecomment-1033865488
[fe89b1b](https://github.com/uBlockOrigin/uAssets/commit/fe89b1b850dc070f09914e36b5520e955a5ad3a6)
@[uBlock-user](https://github.com/uBlock-user) uBlock-user mentioned this issue on 17 Feb
discovery plus italy showing ads #11772
Closed
6 tasks
@https://github.com/uBlockOrigin uBlockOrigin deleted a comment [on 9 Mar](https://github.com/uBlockOrigin/uAssets/issues/1905#event-6207603084)
@[mapx-](https://github.com/mapx-) mapx- mentioned this issue on 25 Mar
hulu.com: ads #12407
Closed
8 tasks
This was referenced on 27 Mar
philo.com: ads #12469
Closed
player.stv.tv: ads #12513
Closed
@[stephenhawk8054](https://github.com/stephenhawk8054) stephenhawk8054 mentioned this issue on 4 Apr
mech-plus.com: ads #12589
Closed
8 tasks
@[stephenhawk8054](https://github.com/stephenhawk8054) stephenhawk8054 mentioned this issue on 4 May
amc.com: ads #13065
Closed
8 tasks
This was referenced on 4 Jul
primevideo.com: ads #13906
Closed
animalplanet.com: ads #13969
Closed
tvbanywherena.com: video ads #13978
Closed
This was referenced 23 days ago
vix.com: ads #14213
Closed
discoveryplus.com: ads #10870
Closed
discoveryplus.com: ads #14365
Closed
@[mapx-](https://github.com/mapx-) mapx- mentioned this issue 6 days ago
go.tlc.com: ads #14451
Closed
8 tasks
@[MasterKia](https://github.com/MasterKia) MasterKia mentioned this issue 5 days ago
primevideo.com: ads #14512

**Comment by nnethercote** [comment]
I have been watching SBS shows with uBlock Origin in Firefox for the past few weeks, and all ads were successfully skipped. Two days ago I started getting the same error mentioned in the description at the top of this issue. I had to disable uBlock Origin for www.sbs.com.au to see the shows. Nothing else about the site has changed, as far as I can tell.

**Comment by stephenhawk8054** [comment]
It was fixed few hours ago. Can you force update all your filter lists and test again?

**Comment by nnethercote** [comment]
It works! I did "Purge all caches" and then "Update now" in the settings pane and it is ad-free again. Thank you!
